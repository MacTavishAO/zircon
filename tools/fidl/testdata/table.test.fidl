library fidl.test.json;

table EmptyTable {
};

table SimpleTable {
    1: int64 x;
    2: reserved;
    3: reserved;
    4: reserved;
    5: int64 y;
};

table OlderSimpleTable {
    1: int64 x;
    2: reserved;
};

table NewerSimpleTable {
    1: int64 x;
    2: reserved;
    3: reserved;
    4: reserved;
    5: int64 y;
    6: int64 z;
    7: reserved;
};

table ReverseOrdinalTable {
    3: int64 x;
    2: int64 y;
    1: int64 z;
};
