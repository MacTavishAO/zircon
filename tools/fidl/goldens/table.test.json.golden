{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [],
  "service_declarations": [],
  "struct_declarations": [],
  "table_declarations": [
    {
      "name": "fidl.test.json/EmptyTable",
      "location": {
        "filename": "table.test.fidl",
        "line": 3,
        "column": 7,
        "length": 10
      },
      "members": [],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SimpleTable",
      "location": {
        "filename": "table.test.fidl",
        "line": 6,
        "column": 7,
        "length": 11
      },
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "x",
          "location": {
            "filename": "table.test.fidl",
            "line": 7,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 8,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 3,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 9,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 4,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 10,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 5,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "table.test.fidl",
            "line": 11,
            "column": 14,
            "length": 1
          }
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 96,
        "has_padding": false,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/OlderSimpleTable",
      "location": {
        "filename": "table.test.fidl",
        "line": 14,
        "column": 7,
        "length": 16
      },
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "x",
          "location": {
            "filename": "table.test.fidl",
            "line": 15,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 16,
            "column": 5,
            "length": 11
          }
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 24,
        "has_padding": false,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/NewerSimpleTable",
      "location": {
        "filename": "table.test.fidl",
        "line": 19,
        "column": 7,
        "length": 16
      },
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "x",
          "location": {
            "filename": "table.test.fidl",
            "line": 20,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 21,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 3,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 22,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 4,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 23,
            "column": 5,
            "length": 11
          }
        },
        {
          "ordinal": 5,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "table.test.fidl",
            "line": 24,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 6,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "z",
          "location": {
            "filename": "table.test.fidl",
            "line": 25,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 7,
          "reserved": true,
          "location": {
            "filename": "table.test.fidl",
            "line": 26,
            "column": 5,
            "length": 11
          }
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 120,
        "has_padding": false,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/ReverseOrdinalTable",
      "location": {
        "filename": "table.test.fidl",
        "line": 29,
        "column": 7,
        "length": 19
      },
      "members": [
        {
          "ordinal": 3,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "x",
          "location": {
            "filename": "table.test.fidl",
            "line": 30,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "table.test.fidl",
            "line": 31,
            "column": 14,
            "length": 1
          }
        },
        {
          "ordinal": 1,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "z",
          "location": {
            "filename": "table.test.fidl",
            "line": 32,
            "column": 14,
            "length": 1
          }
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 72,
        "has_padding": false,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    }
  ],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/SimpleTable",
    "fidl.test.json/ReverseOrdinalTable",
    "fidl.test.json/OlderSimpleTable",
    "fidl.test.json/NewerSimpleTable",
    "fidl.test.json/EmptyTable"
  ],
  "declarations": {
    "fidl.test.json/EmptyTable": "table",
    "fidl.test.json/SimpleTable": "table",
    "fidl.test.json/OlderSimpleTable": "table",
    "fidl.test.json/NewerSimpleTable": "table",
    "fidl.test.json/ReverseOrdinalTable": "table"
  }
}
